pyro_risks/models/score_v0.py

```diff
@@ -116,19 +132,19 @@ def add_lags(df, cols):
     Returns:
         [type]: [description]
     """
-    for var in cols:
-        for dep in df['departement'].unique():
-            tmp = df[df['departement'] == dep][['day', var]].set_index('day')
-            tmp1 = tmp.copy()
-            tmp1 = tmp1.join(tmp.shift(periods=1, freq='D'), rsuffix='_lag1', how='left')
-            tmp1 = tmp1.join(tmp.shift(periods=3, freq='D'), rsuffix='_lag3', how='left')
-            tmp1 = tmp1.join(tmp.shift(periods=7, freq='D'), rsuffix='_lag7', how='left')
-            new_vars = [var + '_lag1', var + '_lag3', var + '_lag7']
-            df.loc[df['departement'] == dep, new_vars] = tmp1[new_vars].values
+    df = df.sort_values(by=["departement", "day"], ascending=False)
+    for feature in cols:
+        df[f"{feature}_lag1"] = df[feature].shift(-1)
+        df[f"{feature}_lag3"] = df[feature].shift(-2)
+        # 4 and not 7 because we only have values of J, J-1, J-3 and J-7
+        # therefore J-7 data is at the 3rd position
+        df[f"{feature}_lag7"] = df[feature].shift(-3)
```

Comment:
I do not agree with the change, this function should work not only for data to predict but in full generality. Indeed this one will apply as well for the creation of the train dataset. If you apply it to data of one entire month say, this function will not create the lags it is meant to create!

Reply:
if you really want only to speed up the predict step, you should maybe rename this function `add_lags_to_predict` or something similar to differentiate it from the former `add_lags`

Reply:
Indeed it does not work with datasets other than those composed of four lines J, J-1, J-3 and J-7 and thus works all for the data to predict indeed. I did not think that we would need this function for something else, since it was only used in the predict pipeline. 
Thus I will rename it accordingly 👍

Reply:
Tell me if it's okay for you now @miltminz